packages/jupyter-ai/jupyter_ai/document_loaders/directory.py

```diff
@@ -136,7 +137,12 @@ def collect_filepaths(path, all_files: bool):
         filepaths = walk_directory(path, all_files)
     else:
         filepaths = []
-        for glob_path in iglob(str(path), include_hidden=all_files, recursive=True):
+        glob_iterator = (
+            iglob(str(path), recursive=True, include_hidden=all_files)
+            if "include_hidden" in signature(iglob).parameters
+            else iglob(str(path), recursive=True)
+        )
+        for glob_path in glob_iterator:
```

Comment:
This silently ignores the `all_files` argument when using glob expressions in Python <3.11. I think we should avoid Python-version-dependent features as much as possible in Jupyter AI, given how confusing they can be. 

Shell globs don't match hidden files by default in both `bash` and `zsh`. Furthermore, support for the `all_files` argument when using glob expressions is not in the user documentation, and therefore this can be considered an implementation detail IMO. Therefore, I think we should recommend users who want to learn hidden files call `/learn .*` instead of `/learn -a ./*`, and just ignore the `-a` argument when a glob expression is specified.

Can you revert these changes and just drop the `include_hidden` argument from the original code to fix this issue while avoiding differing behavior based on Python version?

Reply:
Done in https://github.com/jupyterlab/jupyter-ai/pull/1029/commits/f435a4df69eb2321a66d4ba5eb6978c04e1f4477.